cmd/gen-go-sample/primval.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"strings"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+)
+
+// validPrims contains functions to check value-validity for primitive types.
+// Functions only perform cursory checking; since we use scanner for tokenizing,
+// they must already be a valid token of some type.
+var validPrims = [...]func(string) bool{
```

Comment:
I suggest making the name a verb, since these are functions: `validatePrimitives` or `checkPrimitives`